core/src/replay_stage.rs

```diff
@@ -830,6 +848,46 @@ impl ReplayStage {
         }
     }
 
+    fn retransmit_latest_unpropagated_leader_slot(
+        poh_recorder: &Arc<Mutex<PohRecorder>>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+        retransmit_slots_sender: &RetransmitSlotsSender,
+        progress: &mut ProgressMap,
+    ) {
+        let start_slot = poh_recorder.lock().unwrap().start_slot();
+        if let Some(latest_leader_slot) = progress.get_latest_leader_slot(start_slot) {
+            let bank = bank_forks
+                .read()
+                .unwrap()
+                .get(latest_leader_slot)
+                .expect(
+                    "In order for propagated check to fail, \
+                        latest leader must exist in progress map, and thus also in BankForks",
+                )
+                .clone();
```

Comment:
I realized we don't need to send this bank at all to the retransmit thread. I removed that requirement from the channel here: https://github.com/solana-labs/solana/pull/21943